spark/v3.4/spark/src/test/java/org/apache/iceberg/spark/SparkCatalogConfig.java

```diff
@@ -29,22 +29,27 @@ public enum SparkCatalogConfig {
       SparkCatalog.class.getName(),
       ImmutableMap.of(
           "type", "hive",
-          "default-namespace", "default")),
+          "default-namespace", "default",
+          "unique-table-location", "true")),
   HADOOP(
       "testhadoop",
       SparkCatalog.class.getName(),
       ImmutableMap.of("type", "hadoop", "cache-enabled", "false")),
   REST(
       "testrest",
       SparkCatalog.class.getName(),
-      ImmutableMap.of("type", "rest", "cache-enabled", "false")),
+      ImmutableMap.of(
+          "type", "rest",
+          "cache-enabled", "false",
+          "unique-table-location", "true")),
   SPARK(
       "spark_catalog",
       SparkSessionCatalog.class.getName(),
       ImmutableMap.of(
           "type", "hive",
           "default-namespace", "default",
           "parquet-enabled", "true",
+          "unique-table-location", "true",
```

Comment:
I don't think we should be setting this to true everywhere in this class. Instead, there should be a separate catalog configuration which is then used in `TestCreateTable`

Reply:
what I mean is that you should add a separate catalog configuration such as 
```
SPARK_WITH_UNIQUE_LOCATION(
      "spark_with_unique_location",
      SparkCatalog.class.getName(),
      ImmutableMap.of(
          "type",
          "rest",
          "cache-enabled",
          "false", // Spark will delete tables using v1, leaving the cache out of sync
          "unique-table-location",
          "true"))
```
Then you would only execute the test with this catalog. It's probably best to create a separate test class anyway

Reply:
Done. I extracted my test to TestUniqueLocation.java and added more tests covering important corner cases with overlapping tables.